train.py

```diff
@@ -367,17 +399,20 @@ def load_checkpoint(args, trainer, epoch_itr):
     return False
 
 
-def load_dataset_splits(args, task):
-    task.load_dataset(args.train_subset, combine=True)
-    for split in args.valid_subset.split(','):
-        for k in itertools.count():
-            split_k = split + (str(k) if k > 0 else '')
-            try:
-                task.load_dataset(split_k, combine=False)
-            except FileNotFoundError as e:
-                if k > 0:
-                    break
-                raise e
+def load_dataset_splits(args, task, dataset_splits, epoch):
+    for dataset_split in dataset_splits:
+        if dataset_split == args.train_subset:
+            task.load_dataset(args.train_subset, combine=True, epoch=epoch)
+        else:
+            for valid_sub_split in args.valid_subset.split(','):
+                for k in itertools.count():
```

Comment:
Let's remove the itertools.count loop, it seems to be redundant with the existing loop in the tasks: https://github.com/fairinternal/fairseq-py/blob/master/fairseq/tasks/translation.py#L137